File: src/com/codingbaby/ohmyidea/action/NullCheckInsertAction.java
package com.codingbaby.ohmyidea.action;

import com.intellij.openapi.actionSystem.DataContext;
import com.intellij.openapi.editor.Caret;
import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.editor.actionSystem.EditorAction;
import com.intellij.openapi.editor.actionSystem.EditorActionHandler;
import org.jetbrains.annotations.Nullable;

/**
 *
 *
 */

public class NullCheckInsertAction extends EditorAction {

    public NullCheckInsertAction() {

        super(new EditorActionHandler() {
            @Override
            protected void doExecute(Editor editor, @Nullable Caret caret, DataContext dataContext) {
                int oldOffset = editor.getCaretModel().getOffset();
                editor.getDocument().insertString(oldOffset, " == null");
            }

        });

    }
}

File: src/main/java/com/codingbaby/ohmyidea/action/NullCheckInsertAction.kt
package com.codingbaby.ohmyidea.action

import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.editor.Caret
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.editor.actionSystem.EditorAction
import com.intellij.openapi.editor.actionSystem.EditorActionHandler


class NullCheckInsertAction : EditorAction(object : EditorActionHandler() {
    override fun doExecute(editor: Editor, caret: Caret?, dataContext: DataContext) {
        val oldOffset = editor.caretModel.offset
        editor.document.insertString(oldOffset, " == null")
    }

})

